### PR TITLE
variable value should del after update

### PR DESCRIPTION
**Pull request by amolpati30**
I extended the existing test case to include the following steps:

1. Create an Ansible variable with a default value and an overridden key.
2. Update the value of the Ansible variable at the host level.
3. After updating, delete the value, ensuring that it reverts to the default value.

Dependent PR: https://github.com/SatelliteQE/airgun/pull/1643

### CODE REVIEW

**Comment by amolpati30** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host
airgun: 1643

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 9422
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host --external-logging
Test Result : ========== 1 passed, 10 deselected, 24 warnings in 817.03s (0:13:37) ===========
```

**Review by ColeHiggins2 — APPROVED** [review]

**Comment by amolpati30** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host
airgun: 1643

File: tests/foreman/ui/test_ansible.py [hunk 1]
```diff
@@ -151,21 +151,28 @@ def test_positive_assign_ansible_role_variable_on_host(
...
+
         :customerscenario: true
 
         :steps:
             1. Create an Ansible variable with array type and set the default value.
             2. Enable both 'Merge Overrides' and 'Merge Default'.
             3. Add the variable to a role and attach the role to the host.
             4. Verify that ansible role and variable is added to the host.
+            5. In variable override the value.
+            6. Delete the overridden value.
```
Comment (shweta83) [inline tests/foreman/ui/test_ansible.py:170]:
```suggestion
            6. Reset the overridden value to the original value.
```

File: tests/foreman/ui/test_ansible.py [hunk 2]
```diff
@@ -229,6 +236,27 @@ def _finalize():
...
+            session.ansiblevariables.create_with_overrides(
+                {
+                    'key': new_key,
+                    'ansible_role': SELECTED_ROLE[0],
+                    'override': 'true',
+                    'parameter_type': parameter_type,
+                    'default_value': default_value,
+                }
+            )
+            new_value = '["test update"]'
```
Comment (shweta83) [inline tests/foreman/ui/test_ansible.py:249]: Doesn't this give you error?
You can try new_value = '[\"test update\"]'

File: tests/foreman/ui/test_ansible.py [hunk 3]
```diff
@@ -229,6 +236,27 @@ def _finalize():
...
+            # Returns the updated value stored in a variable.
+            update_value = session.host_new.read_update_value(
+                rhel_contenthost.hostname, new_key, new_value
+            )
+            assert new_value in update_value
+            # Revert the updated value to its default state.
+            delete_value = session.host_new.read_value_after_del(rhel_contenthost.hostname, new_key)
```
Comment (shweta83) [inline tests/foreman/ui/test_ansible.py:256]: Can you please rename this variable? The name is confusing.
may be `reset_variable_value` ?

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 9443
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host --external-logging
Test Result : ========== 1 failed, 10 deselected, 22 warnings in 793.28s (0:13:13) ===========
```

File: tests/foreman/ui/test_ansible.py [hunk 4]
```diff
@@ -151,21 +151,28 @@ def test_positive_assign_ansible_role_variable_on_host(
...
+        :Verifies: SAT-23109
+
         :customerscenario: true
 
         :steps:
             1. Create an Ansible variable with array type and set the default value.
             2. Enable both 'Merge Overrides' and 'Merge Default'.
             3. Add the variable to a role and attach the role to the host.
             4. Verify that ansible role and variable is added to the host.
+            5. In variable override the value.
```
[the remark this answers is not in the document]
Comment (amolpati30) [reply tests/foreman/ui/test_ansible.py]: Done.

File: tests/foreman/ui/test_ansible.py (hunk 1 above)
Comment (amolpati30) [reply tests/foreman/ui/test_ansible.py]: Done.

File: tests/foreman/ui/test_ansible.py (hunk 2 above)
Comment (amolpati30) [reply tests/foreman/ui/test_ansible.py]: It did not produce any errors, but I made a small change.

File: tests/foreman/ui/test_ansible.py (hunk 3 above)
Comment (amolpati30) [reply tests/foreman/ui/test_ansible.py]: updated the variable name.

**Comment by amolpati30** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host
airgun: 1643

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 9446
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host --external-logging
Test Result : ========== 1 passed, 10 deselected, 24 warnings in 933.30s (0:15:33) ===========
```

**Review by LadislavVasina1 — APPROVED** [review]

**Review by synkd — APPROVED** [review]

**Comment by amolpati30** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host
airgun: 1643

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 9480
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host --external-logging
Test Result : ========== 1 failed, 10 deselected, 26 warnings in 843.76s (0:14:03) ===========
```

**Comment by amolpati30** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host
airgun: 1643

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 9482
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_ansible.py::TestAnsibleCfgMgmt -k test_positive_assign_ansible_role_variable_on_host --external-logging
Test Result : ========== 1 passed, 10 deselected, 26 warnings in 1629.27s (0:27:09) ==========
```

**Review by shweta83 — APPROVED** [review]
Ack

**Pull request merged**